src/soilbiogeochem/SoilBiogeochemDecompCascadeMIMICSMod.F90

```diff
@@ -0,0 +1,1274 @@
+               spinup_geogterm_m1(c) = spinup_factor(i_cop_mic) * get_spinup_latitude_term(grc%latdeg(col%gridcell(c)))
+            else
+               spinup_geogterm_m1(c) = 1._r8
+            endif
+            !
+            if ( abs(spinup_factor(i_oli_mic) - 1._r8) .gt. .000001_r8) then
+               spinup_geogterm_m2(c) = spinup_factor(i_oli_mic) * get_spinup_latitude_term(grc%latdeg(col%gridcell(c)))
+            else
+               spinup_geogterm_m2(c) = 1._r8
+            endif
+            !
+         end do
+      else
+         do fc = 1,num_soilc
+            c = filter_soilc(fc)
+            spinup_geogterm_l1(c) = 1._r8
+            spinup_geogterm_l2(c) = 1._r8
+            spinup_geogterm_cwd(c) = 1._r8
+            spinup_geogterm_s1(c) = 1._r8
+            spinup_geogterm_s2(c) = 1._r8
+            spinup_geogterm_s3(c) = 1._r8
+            spinup_geogterm_m1(c) = 1._r8
+            spinup_geogterm_m2(c) = 1._r8
+         end do
+      endif
+
+      !--- time dependent coefficients-----!
+      if ( nlevdecomp .eq. 1 ) then
+
+         ! calculate function to weight the temperature and water potential scalars
+         ! for decomposition control.  
+
+
+         ! the following normalizes values in fr so that they
+         ! sum to 1.0 across top nlevdecomp levels on a column
+         frw(bounds%begc:bounds%endc) = 0._r8
+         allocate(fr(bounds%begc:bounds%endc,nlev_soildecomp_standard))
+         do j=1,nlev_soildecomp_standard
+            do fc = 1,num_soilc
+               c = filter_soilc(fc)
+               frw(c) = frw(c) + col%dz(c,j)
+            end do
+         end do
+         do j = 1,nlev_soildecomp_standard
+            do fc = 1,num_soilc
+               c = filter_soilc(fc)
+               if (frw(c) /= 0._r8) then
+                  fr(c,j) = col%dz(c,j) / frw(c)
+               else
+                  fr(c,j) = 0._r8
+               end if
+            end do
+         end do
+
+         ! calculate the rate constant scalar for soil water content.
+         ! Uses the log relationship with water potential given in
+         ! Andren, O., and K. Paustian, 1987. Barley straw decomposition in the field:
+         ! a comparison of models. Ecology, 68(5):1190-1200.
+         ! and supported by data in
+         ! Orchard, V.A., and F.J. Cook, 1983. Relationship between soil respiration
+         ! and soil moisture. Soil Biol. Biochem., 15(4):447-453.
+
+         do j = 1,nlev_soildecomp_standard
+            do fc = 1,num_soilc
+               c = filter_soilc(fc)
+               if (j==1) w_scalar(c,:) = 0._r8
+               psi = min(soilpsi(c,j),maxpsi)
+               ! decomp only if soilpsi is higher than minpsi
+               if (psi > minpsi) then
+                  w_scalar(c,1) = w_scalar(c,1) + (log(minpsi/psi)/log(minpsi/maxpsi))*fr(c,j)
+               end if
+            end do
+         end do
+
+         ! TODO May need a hook from MIMICS to ch4 code for this to work
```

Comment:
@wwieder 
I'm trying to diagnose a `CH4 Conservation Error in CH4Mod during diffusion` when I run with MIMICS.
For now I'm writing out key variables and comparing to a case without MIMICS.
Remembering though that you made the comment that I wrote in line 998, maybe you have some intuition about where to look for the problem?

Reply:
hmm, I'm not really sure what's happening in this code block, but wonder if there is information that's being passed to `CH4Mod` with BGC that needs to be included for MIMICS (e.g.,  https://github.com/slevisconsulting/ctsm/blob/3ff27f32c0d9d5fcfda497f92d2eab0df3403c85/src/biogeochem/ch4Mod.F90#L2344). 

Would it be helpful to discuss next week?

Reply:
It wouldn't hurt and meanwhile I will go back to investigating on Monday morning. This far I have traced the problem to very large values of `source` in `ch4Mod.F90` in my MIMICS case (not so in the default case).

Best days for me next week are Monday, Tuesday, and Friday. Feel free to send me an invite.